akka-stream/src/main/scala/akka/stream/scaladsl/Tcp.scala

```diff
@@ -89,10 +96,9 @@ object Tcp extends ExtensionId[Tcp] with ExtensionIdProvider {
 final class Tcp(system: ExtendedActorSystem) extends akka.actor.Extension {
   import Tcp._
 
-  private val settings = ActorMaterializerSettings(system)
-
   // TODO maybe this should be a new setting, like `akka.stream.tcp.bind.timeout` / `shutdown-timeout` instead?
-  val bindShutdownTimeout = settings.subscriptionTimeoutSettings.timeout
+  val bindShutdownTimeout =
+    system.settings.config.getDuration("akka.stream.materializer.subscription-timeout.timeout").toMillis.millis
```

Comment:
:+1: while 'reaching into" the config like this is not super nice, indeed I think I like it over using `ActorMaterializerSettings`. Another reason to move the setting to `akka.stream.tcp.bind.timeout` as commented.